### PR TITLE
Split the docs generation into a separate module

### PR DESCRIPTION
**Pull request by z8v**
Related issue: https://github.com/tweag/rules_nixpkgs/issues/264

### CODE REVIEW

**Comment by evertedsphere** [comment]
@z8v CI should be fixed as of #297: could you rebase on `master`?

**Comment by avdv** [comment]
@mergifyio rebase

**Comment by avdv** [comment]
:point_up: @z8v / @evertedsphere you can have mergify rebase for you

**Comment by evertedsphere** [comment]
TIL

File: .github/workflows/workflow.yaml [hunk 1]
```diff
@@ -8,6 +8,14 @@ on:
...
 
 jobs:
+  cancel-running-workflows:
+    name: Cancel running workflows
+    runs-on: ubuntu-20.04
+    steps:
+      - name: cancel running workflows
+        uses: styfle/cancel-workflow-action@0.11.0
+        with:
+          access_token: ${{ github.token }}
```
Comment (avdv) [inline .github/workflows/workflow.yaml:18]: Nice! Usually I tried to cancel these manually, when I was aware of them. 

But could you use `ubuntu-latest` please, so the jobs are run on the same image?

And I would prefer this to be split out into a separate commit.

File: core/MODULE.bazel [hunk 2]
```diff
@@ -5,5 +5,3 @@ module(
 
 bazel_dep(name = "platforms", version = "0.0.4")
 bazel_dep(name = "bazel_skylib", version = "1.0.3")
-bazel_dep(name = "rules_sh", version = "0.3.0")
-bazel_dep(name = "stardoc", version = "0.5.1", repo_name = "io_bazel_stardoc", dev_dependency = True)
```
Comment (avdv) [inline core/MODULE.bazel:9]: I am wondering if this is actually correct to remove these lines. Actually, this module still depends on these other modules since it still loads the dependencies from the new location.

**Review by avdv — CHANGES_REQUESTED** [review]
Thank you, that is going into the right direction.

File: .github/workflows/workflow.yaml (hunk 1 above)
Comment (z8v) [reply .github/workflows/workflow.yaml:18]: Switched to `ubuntu-latest` and added a separate commit.

File: core/MODULE.bazel (hunk 2 above)
Comment (z8v) [reply core/MODULE.bazel:9]: Yes, it seems like unless the tests are a separate module we'd always need to bring those in.

File: docs/MODULE.bazel [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module(
+    name = "docs_generation",
```
Comment (aherrmann) [inline docs/MODULE.bazel:2]: To not collide with other modules on the BCR this should use a more specific module name, e.g. `rules_nixpkgs_docs` to follow the pattern of the other modules.

File: docs/MODULE.bazel [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module(
```
Comment (aherrmann) [inline docs/MODULE.bazel:1]: With this becoming a module I think we should also add a corresponding WORKSPACE for the legacy setup.

File: docs/stardoc.bzl [hunk 5]
```diff
@@ -53,11 +53,11 @@ def stardoc(
         name,
         out,
         input,
-        aspect_template = "@rules_nixpkgs_core//docs:templates/aspect.vm",
-        func_template = "@rules_nixpkgs_core//docs:templates/func.vm",
-        header_template = "@rules_nixpkgs_core//docs:templates/header.vm",
-        provider_template = "@rules_nixpkgs_core//docs:templates/provider.vm",
-        rule_template = "@rules_nixpkgs_core//docs:templates/rule.vm",
+        aspect_template = "@io_tweag_rules_nixpkgs//docs:templates/aspect.vm",
```
Comment (aherrmann) [inline docs/stardoc.bzl:56]: I think these should become references to the docs module, e.g. `@rules_nixpkgs_docs//:templates/rule.vm`.

**Review by aherrmann — CHANGES_REQUESTED** [review]
I added a couple more comments. I think we should be consistent in the module and legacy WORKSPACE worlds to avoid labels that work in only one of the worlds.

File: WORKSPACE [hunk 6]
```diff
@@ -2,15 +2,20 @@ workspace(name = "io_tweag_rules_nixpkgs")
 
 # For documentation
 
+local_repository(
+    name = "io_tweag_rules_nixpkgs_docs",
```
Comment (aherrmann) [inline WORKSPACE:6]: With the modularization we also wanted to drop the `io_tweag_` prefix in the individual modules.
```suggestion
    name = "rules_nixpkgs_docs",
```
IIRC it was only still kept in the top-level `io_tweag_rules_nixpkgs` for backward compatibility.
All the other modules, e.g. `rules_nixpkgs_java`, don't use that prefix anymore.

File: core/MODULE.bazel [hunk 7]
```diff
@@ -5,5 +5,3 @@ module(
 
 bazel_dep(name = "platforms", version = "0.0.4")
 bazel_dep(name = "bazel_skylib", version = "1.0.3")
```
Comment (aherrmann) [inline core/MODULE.bazel:7]:
```suggestion
bazel_dep(name = "bazel_skylib", version = "1.0.3")
bazel_dep(name = "rules_sh", version = "0.3.0", dev_dependency = True)
```
IIUC `stardoc` is indeed no longer a dependency of `rules_nixpkgs_core`. However, `rules_sh` is still needed for the tests, but, not for regular usage. In that case it should be marked as a dev-dependency.

**Review by aherrmann — APPROVED** [review]
One last small comment about the rules_sh dev-dependency.
I also noticed that the docs regeneration comment in the README was out of date and pushed an update.
Apart from that this looks ready to me. Thank you for pushing this forward!

**Review by avdv — APPROVED** [review]
Thank you!
